MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
+                                         boolean isPirate) {
+
+        // Get all valid bomber
+        List<Entity> bomberList = entityList.stream().filter(Targetable::isBomber).collect(Collectors.toList());
+        if (bomberList.isEmpty()) {
+            return;
+        }
+
+        int minThrust;
+        int maxLoad;
+        int maxBombers = (int) Math.ceil(((Compute.randomInt(60) + 40) / 100.0) * bomberList.size());
         int numBombers = 0;
 
-        int[] validBombChoices = groundMap ? validBotBombs : validBotAABombs;
+        int[] generatedBombs = new int[BombType.B_NUM];
+        Map<Integer, int[]> bombsByCarrier = new HashMap<>();
 
-        for (Entity entity : entityList) {
-            if (entity.isBomber()) {
-                // if this entity has no guns (e.g. is a Boeing Jump Bomber)
-                if (entity.getIndividualWeaponList().isEmpty()) {
-                    loadBombs(entity, validBombChoices, campaign.getGameYear());
-                    continue;
+        boolean hasGuided = false;
+        for (int i = 0; i < bomberList.size(); i++) {
+            if (numBombers >= maxBombers) {
+                break;
+            }
+
+            Entity curBomber = bomberList.get(i);
+            boolean isUnarmed = curBomber.getIndividualWeaponList().isEmpty();
+
+            // Some fighters on ground attack may be flying air cover rather than strictly
+            // air-to-ground
+            boolean isCAP = !hasGroundTargets || (Compute.d6() == 1);
+
+            // Set minimum thrust values, with lower minimums for unarmed and ground attack,
+            // and use remaining thrust to limit hardpoints
+            if (isCAP) {
+                minThrust = isUnarmed ? 2 : ((int) Math.ceil(curBomber.getWalkMP() / 2.0));
+            } else {
+                minThrust = isUnarmed ? 2 : 3;
+            }
+            maxLoad = Math.min((int) Math.floor(curBomber.getWeight() / 5.0), (curBomber.getWalkMP() - minThrust) * 5);
+
+            // Get a random percentage (40 - 90) of the maximum bomb load for armed entities
+            if (!isUnarmed) {
+                maxLoad = (int) Math.ceil((Compute.randomInt(50) + 40) * maxLoad / 100.0);
+            }
+
+            if (maxLoad == 0) {
+                continue;
+            }
+
+            // Generate bomb load
+            generatedBombs = generateExternalOrdnance(
+                    maxLoad,
+                    isCAP,
+                    isPirate,
+                    quality,
+                    campaign.getGameYear());
+            // Whoops, go yell at the ordnance technician
+            if (Arrays.stream(generatedBombs).sum() == 0) {
+                continue;
+            }
+
+            // Set a flag if any of the ordnance guided by TAG
+            hasGuided = !hasGuided &&
+                    (generatedBombs[BombType.B_LG] > 0 ||
+                            generatedBombs[BombType.B_HOMING] > 0);
```

Comment:
See comment below; instead of two separate locations with different tests of hard-coded munitions, we should have a set or list of guided munitions that we can use to check for matches, and it should include ASF Homing Arrow IV bomb munitions as well.